mongo/pagination/pagination_by_cursor.go

```diff
@@ -0,0 +1,103 @@
+package pagination
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/Scalingo/go-utils/mongo/document"
+	"github.com/pkg/errors"
+	"gopkg.in/mgo.v2/bson"
+)
+
+func NewByCursorService(opts ServiceOpts) ServiceByCursor {
+	return opts
+}
+
+type PaginateByCursorOpts struct {
+	Cursor      bson.M
+	AmountItems int
+	Query       bson.M
+	SortOrder   string
+}
+
+type ServiceByCursor interface {
+	PaginateByCursor(ctx context.Context, collection string, result interface{}, opts PaginateByCursorOpts) error
+}
+
+func (s ServiceOpts) paramValidationByCursor(collection string, opts *PaginateByCursorOpts) error {
```

Comment:
With my previous comment, `opts` shouldn't be a pointer anymore:

```suggestion
func (s ServiceOpts) paramValidationByCursor(collection string, opts PaginateByCursorOpts) error {
```